pkg/devfile/validate/events.go

```diff
@@ -47,21 +61,15 @@ func validateEvents(events common.DevfileEvents, commands map[string]common.Devf
 	return nil
 }
 
-// isEventValid checks if events belonging to a specific event type are valid:
-// 1. Event should map to a valid devfile command
-// 2. Event commands should be valid
+// isEventValid checks if events belonging to a specific event type are valid ie; either exec or composite command
 func isEventValid(eventNames []string, eventType string, commands map[string]common.DevfileCommand, components []common.DevfileComponent) error {
 	eventErrorMsg := make(map[string][]string)
 	eventErrors := ""
 
 	for _, eventName := range eventNames {
-		isEventPresent := false
 		for _, command := range commands {
-			// Check if event matches a valid devfile command
 			if command.GetID() == strings.ToLower(eventName) {
-				isEventPresent = true
-
-				// Check if the devfile command is valid
+				// Check if the devfile command is a valid odo devfile command
 				err := validateCommand(command, commands, components)
```

Comment:
Since the `commands` map contain all the commands in the devfile and they are all validated before this call, can we just check if a `eventName` is in the `commands` map instead of validating each event again? What am I missing?

Reply:
good point

Reply:
updated!, pls help review!